src/server/pfs/server/diff.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset"
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+// Differ compares two sources and iterates over the items that are not equal.
+type Differ struct {
+	a, b Source
+}
+
+func NewDiffer(a, b Source) *Differ {
+	return &Differ{a: a, b: b}
+}
+
+func (d *Differ) IterateDiff(ctx context.Context, cb func(aFi, bFi *pfs.FileInfoV2) error) error {
```

Comment:
Seems like calling this function `Iterate` would be more appropriate considering the abstraction is specifically for diffing.